src/ui/components/sdcard.c

```diff
@@ -19,22 +19,41 @@
 #include <hardfault.h>
 #include <screen.h>
 #include <sd.h>
-#include <string.h>
 #include <touch/gestures.h>
 #include <ui/screen_stack.h>
 
 typedef struct {
-    // if true, the callback won't be called until the sd card is inserted.
-    // the insert/remove label changes depending on this flag.
-    bool insert;
     void (*callback)(bool, void*);
     void* callback_param;
+    // TODO: use a TIMER interrupt to get a more accurate timer.
+    // 250 is ~0.5 sec. Unit: rendering rate.
+    int check_interval;
+    int count;
 } data_t;
 
+static void _insert_poll_callback(component_t* component)
+{
+    data_t* data = (data_t*)component->data;
+    if (sd_card_inserted()) {
+        if (data->callback) {
```

Comment:
`if(data->callback && sd_card_inserted())` ? not sure if `sd_card_inserted()` is cheap or expensive, but if expensive, it's better to call it only when needed.